prov/psm2/src/psmx2.h

```diff
@@ -955,6 +955,39 @@ void	psmx2_ns_add_local_name(int service, struct psmx2_ep_name *name);
 void	psmx2_ns_del_local_name(int service, struct psmx2_ep_name *name);
 void	*psmx2_ns_resolve_name(const char *server, int *service);
 
+static inline
+void	psmx2_ep_name_to_str(struct psmx2_ep_name *ep_name, char *str_name)
+{
+	memcpy(str_name, ep_name, sizeof(*ep_name));
+	str_name[sizeof(*ep_name)] = '\0';
+}
+static inline
+void	psmx2_str_to_ep_name(const char *str_name, struct psmx2_ep_name *ep_name)
+{
+	memcpy(ep_name, str_name, sizeof(*ep_name));
+}
+static inline
+int	psmx2_ns_ep_name_compare(const char *ep_name_str1, const char *ep_name_str2)
+{
```

Comment:
It's better to have the utility code take the raw address directly. The following conversion can be avoided.